ecl/resource_ecl_sss_approval_request_v1.go

```diff
@@ -0,0 +1,182 @@
+						},
+						"text": &schema.Schema{
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+			"request_user": &schema.Schema{
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"approver": &schema.Schema{
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"approval_deadline": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"approval_expire": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"registered_time": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"updated_time": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"status": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"approved", "denied", "cancelled",
+				}, false),
+			},
+		},
+	}
+}
+
+func resourceSSSApprovalRequestV1Create(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	client, err := config.sssV1Client(GetRegion(d, config))
+	if err != nil {
+		return fmt.Errorf("Error creating ECL sss client: %s", err)
+	}
+
+	opts := approval_requests.UpdateOpts{Status: d.Get("status").(string)}
+	log.Printf("[DEBUG] Update Options: %#v", opts)
+
+	approval, err := approval_requests.Update(client, d.Get("request_id").(string), opts).Extract()
+	if err != nil {
+		return fmt.Errorf("error updating ECL approval request: %s", err)
+	}
+
+	d.SetId(d.Get("request_id").(string))
+
+	log.Printf("[DEBUG] Updated ECL approval request %s: %#v", approval.RequestID, approval)
+	return resourceSSSApprovalRequestV1Read(d, meta)
+}
+
+func resourceSSSApprovalRequestV1Read(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	client, err := config.sssV1Client(GetRegion(d, config))
+	if err != nil {
+		return fmt.Errorf("error creating ECL sss client: %s", err)
+	}
+
+	approval, err := approval_requests.Get(client, d.Id()).Extract()
+	if err != nil {
+		return CheckDeleted(d, err, "approval request")
+	}
+	log.Printf("[DEBUG] Retrieved ECL approval request: %#v", approval)
+
+	d.Set("request_id", approval.RequestID)
+	d.Set("external_request_id", approval.ExternalRequestID)
+	d.Set("approver_type", approval.ApproverType)
+	d.Set("approver_id", approval.ApproverID)
+	d.Set("request_user_id", approval.RequestUserID)
+	d.Set("service", approval.Service)
+	d.Set("actions", approval.Actions)
+	d.Set("descriptions", approval.Descriptions)
+	d.Set("request_user", approval.RequestUser)
+	d.Set("approver", approval.Approver)
+	d.Set("approval_deadline", approval.ApprovalDeadLine)
+	d.Set("approval_expire", approval.ApprovalExpire)
+	d.Set("registered_time", approval.RegisteredTime)
+	d.Set("updated_time", approval.UpdatedTime)
+	d.Set("status", approval.Status)
+
+	return nil
+}
+
+func resourceSSSApprovalRequestV1Update(d *schema.ResourceData, meta interface{}) error {
```

Comment:
I think you do not need to define it. The same applies to `resourceSSSApprovalRequestV1Delete`.

Reply:
You can remove it if the following issues are resolved.
https://github.com/nttcom/terraform-provider-ecl/pull/46#discussion_r376225289